multiqc/modules/bbmap/bbmap.py

```diff
@@ -69,6 +68,42 @@ def __init__(self):
                     plot = self.make_basic_table(file_type)
                 )
 
+        self.qchist_data = dict()
+        headers = OrderedDict()
+
+        for f in self.find_log_files('bbmap/qchist'):
+            self.qchist_data[f['s_name']] = dict()
+            my_total=[]
+            for l in f['f'].splitlines()[1:]:
+                splitted=l.split("\t", 3)
+                key=splitted[0]
+                if int(key) >= 30:
+                    value = float(splitted[2])
+                    my_total.append(value)
+                self.qchist_data[f['s_name']][key] = sum(my_total)*100
+            headers[key] = {
+                        'title': '% Q30 bases',
+                        'description': 'BBMap qchist - Percentage of bases with QUAL>=30',
+                        'suffix': ' %',
+                        'scale': 'RdYlGn-rev',
+                        'format': '{:,.2f}'
+            }
+        self.general_stats_addcols(self.qchist_data, headers)
+
+        # Filter out samples matching ignored sample names
+        self.qchist_data = self.ignore_samples(self.qchist_data)
+
+        # Nothing found - raise a UserWarning to tell MultiQC
+        if len(self.qchist_data) == 0:
+            log.debug("Could not find any reports in {}".format(config.analysis_dir))
+            raise UserWarning
```

Comment:
This will stop the bbmap module from generating any report at all if no qchist data is parsed. It has to be handled as part of the bbmap module as a whole.

Reply:
Fixed in a8d98285faf4be1d6ab3577434d60169f2ad651a